src/main/java/caculator/PositiveOrZero.java

```diff
@@ -0,0 +1,22 @@
+package caculator;
+
+public record PositiveOrZero(int value) {
+
+    public PositiveOrZero {
+        validateInvariant(value);
+    }
+
+    public PositiveOrZero(String text) {
+        this(Integer.parseInt(text));
+    }
+
+    private void validateInvariant(int value) {
+        if (isNegative(value)) {
+            throw new IllegalArgumentException("음수는 입력할 수 없습니다: " + value);
+        }
+    }
+
+    private boolean isNegative(int number) {
+        return number < 0;
+    }
```

Comment:
```suggestion
    private boolean isNegative() {
        return this.value < 0;
    }
```

PositiveOrZero가 값을 가지고 있는데 굳이 메서드 인자로 전달할 필요가 있을까?